pkg/util/file_indexer.go

```diff
@@ -83,6 +85,75 @@ func resolveFilePath(directory string) (string, error) {
 	return directory, nil
 }
 
+// gitignoreFilePath gives the filepath of the .gitignore file in the contex
+func gitIgnoreFilePath(directory string) (string, error) {
+	directoryFile, err := os.Stat(filepath.Join(directory))
+	if err != nil {
+		return "", err
+	}
+
+	switch mode := directoryFile.Mode(); {
+	case mode.IsDir():
+		// whether the mode is a directory stuff
+		// mode directory bit is set
+		return filepath.Join(directory, ".gitignore"), nil
+	case mode.IsRegular():
+		// whether the mode is a file stuff
+		// No mode type is set
+		return filepath.Join(filepath.Dir(directory), ".gitignore"), nil
+	}
+
+	return directory, nil
+}
+
+// Check .gitignore file exists or not
+// Add odo-file-index.json entry to .gitignore
+func checkGitIgnoreFile(directory string) error {
+
+	ignoreFile, err := gitIgnoreFilePath(directory)
+	_, err = os.Stat(ignoreFile)
+
+	// .odo/odo-file-index.json file should be created in .odo
+	// open, read and write .odo/odo-file-index.json to .gitignore file if exists
+	// if .gitignore does not exist then create the file and write .odo/odo-file-index.json to it
+	if err == nil {
+		file, err := os.OpenFile(ignoreFile, os.O_APPEND|os.O_RDWR, 0600)
+		if err != nil {
+			return errors.Wrap(err, "failed opening file")
+		}
+		defer file.Close()
+
+		data, err := ioutil.ReadFile(ignoreFile)
+		if err != nil {
+			return errors.Wrap(err, "failed reading data from file")
+		} else {
```

Comment:
Like I mentioned in previous comment, in Go, we generally have a check for error and then the code that is supposed to be executed if there was no error. So in this case, we could simplify it to:
```go
data, err := ioutil.ReadFile(ignoreFile)
if err != nil {
	return errors.Wrap(err, "failed reading data from file")
}
// check whether .odo/odo-file-index.json is already in the .gitignore file
if !strings.Contains(string(data), ".odo/odo-file-index.json") {
	if _, err := file.WriteString("\n" + ".odo/odo-file-index.json"); err != nil {
			return errors.Wrap(err, "failed writing string to file")
	}
}
```